### PR TITLE
Suscat tani

### PR DESCRIPTION
**Pull request by TBarasch**
fixed function, added test file

### CODE REVIEW

File: tests/test-suscat.R [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+testthat::test_that("Output of wrong type",{
+        testthat::expect_type(suscat(df,c('x')),"list")
+})
+
+#testing output
+testthat::test_that("Output unexpectedly wrong",{
+        #checking indicies per column
+        testthat::expect_equal(length(suscat(df,c('x')$'x')),2)
+        #checking number of columns
+        testthat::expect_equal(length(suscat(df,c('x'),n=0)),1)
```
Comment (EitharAlfatih) [inline tests/test-suscat.R]: I am getting an error here

File: R/suscat.R [hunk 2]
```diff
@@ -1,18 +1,59 @@
...
 #' @examples
 #' suscat(df, col = c(1,3,5), n = 10)
-#' suscat(df, col = c(1,5,4,10), n = 0.02)
-suscat  <- function(df, col = NULL, n = 0.01){
-  
-  
-  return()
-}
\ No newline at end of file
+#' suscat(df, col = c(1,5,4,10), n = 2, num = 'number')
```
Comment (EitharAlfatih) [inline R/suscat.R]: The result of the examples can be added

**Review by EitharAlfatih — CHANGES_REQUESTED** [review]

File: tests/testthat/test-suscat.R [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+testthat::test_that("Output unexpectedly wrong",{
+        #checking indicies per column
+        testthat::expect_equal(length(suscat(df,c('x')$'x')),2)
+        #checking number of columns
+        testthat::expect_equal(length(suscat(df,c('x'),n=0)),1)
+        # check values of indicies returned
+        testthat::expect_gte(min(suscat(df,c('x')$'x')),0)
+        testthat::expect_lte(max(suscat(df,c('x')$'x')),nrow(df))
+        testthat::expect_equal(suscat(df,c('x','y'))$'x',suscat(df,c('x'))$'x')
+})
```
Comment (EitharAlfatih) [inline tests/testthat/test-suscat.R]: these are failing in my laptop.

**Review by EitharAlfatih — CHANGES_REQUESTED** [review]

File: tests/testthat/test-suscat.R [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+#testing output
+testthat::test_that("Output unexpectedly wrong",{
+        #checking indicies per column
+        testthat::expect_equal(length(suscat(df,c('x')$'x')),2)
+        #checking number of columns
+        testthat::expect_equal(length(suscat(df,c('x'),n=0)),1)
+        # check values of indicies returned
+        testthat::expect_gte(min(suscat(df,c('x')$'x')),0)
+        testthat::expect_lte(max(suscat(df,c('x')$'x')),nrow(df))
+        testthat::expect_equal(suscat(df,c('x','y'))$'x',suscat(df,c('x'))$'x')
```
Comment (EitharAlfatih) [inline tests/testthat/test-suscat.R]:
```suggestion
        #checking indicies per column
        testthat::expect_equal(length(suscat(df,c('x'))$'x'),2)
        #checking number of columns
        testthat::expect_equal(length(suscat(df,c('x'),n=0)),1)
        # check values of indicies returned
        testthat::expect_gte(min(suscat(df,c('x'))$'x'),0)
        testthat::expect_lte(max(suscat(df,c('x'))$'x'),nrow(df))
        testthat::expect_equal(suscat(df,c('x','y'))$'x',suscat(df,c('x'))$'x')
```

File: tests/test-suscat.R (hunk 1 above)
Comment (TBarasch) [reply tests/test-suscat.R]: corrected in new version

**Review by EitharAlfatih — APPROVED** [review]

**Comment by EitharAlfatih** [comment]
tested. Things are working fine

**Pull request merged**